ark/api/v1alpha1/query_types.go

```diff
@@ -32,9 +53,15 @@ type EvaluatorRef struct {
 }
 
 type QuerySpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=user;messages
+	// +kubebuilder:default=user
+	Type string `json:"type,omitempty"`
```

Comment:
@havasik should we do the admission conversion to `user` by default so that we always have a value, consistent with what `model` will be for query (and `type` `ClusterIP` for services, if i remember correctly)